src/reconcile/rec_write.c

```diff
@@ -1113,12 +1119,13 @@ __rec_child_modify(WT_SESSION_IMPL *session,
 			 * process will have completed before we walk any pages
 			 * for checkpoint.
 			 */
-			if ((ret = __wt_page_in(session, ref,
-			    WT_READ_CACHE | WT_READ_NO_EVICT |
-			    WT_READ_NO_GEN | WT_READ_NO_WAIT)) == WT_NOTFOUND) {
+			ret = __wt_page_in(session, ref,
+			    WT_READ_CACHE | WT_READ_NO_EVICT | WT_READ_NO_GEN);
+			if (ret == WT_NOTFOUND) {
```

Comment:
@michaelcahill, I'm missing the point of this change: `WT_READ_NO_WAIT` would cause immediate return from `__wt_page_in` if the ref state is `WT_REF_READING` or `WT_REF_LOCKED`, which makes sense to me here, what am I missing?